nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/GetSolr.java

```diff
@@ -172,157 +203,196 @@ protected void init(final ProcessorInitializationContext context) {
 
     @Override
     public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
-        lastEndDatedRef.set(UNINITIALIZED_LAST_END_DATE_VALUE);
+        clearState.set(true);
```

Comment:
Probably we'd like to clear state only when following properties get changed? It would be a bad UX if state is cleared when user re-configure batch size.
- SOLR_TYPE
- SOLR_LOCATION
- COLLECTION
- SOLR_QUERY
- DATE_FIELD
- RETURN_FIELDS

Reply:
ok, no problem